src/lib/generate-output.ts

```diff
@@ -16,15 +16,16 @@ interface BasicOutput {
 	filename: string | undefined;
 }
 
+interface GenerateOutputProps {
+	html: string;
+	relativePath: string;
+	config: PdfConfig | HtmlConfig | Config;
+	browser: puppeteer.BrowserContext;
+}
 /**
  * Generate the output (either PDF or HTML).
  */
-export async function generateOutput(html: string, relativePath: string, config: PdfConfig): Promise<PdfOutput>;
-export async function generateOutput(html: string, relativePath: string, config: HtmlConfig): Promise<HtmlOutput>;
-export async function generateOutput(html: string, relativePath: string, config: Config): Promise<Output>;
-export async function generateOutput(html: string, relativePath: string, config: Config): Promise<Output> {
-	const browser = await puppeteer.launch({ devtools: config.devtools, ...config.launch_options });
-
+export async function generateOutput({ html, relativePath, config, browser }: GenerateOutputProps): Promise<Output> {
```

Comment:
I'm not sure you understood the purpose of these overloads? They give better types for certain inputs, e.g. if you pass a `PdfConfig` as `config` you'd get a `PdfOutput` result which is more specific than `Output`.

Reply:
Yes, you are right.
Thanks for the explanation, didn't saw the different return types.